sycl/test-e2e/KernelCompiler/opencl_queries.cpp

```diff
@@ -0,0 +1,47 @@
+//==------- ocloc_queries.cpp ----------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: ocloc
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Note: the ocloc queries that are tested here are relatively new ( Dec 2023 )
+// if encountering many failures then an outdated ocloc is the likely culprit.
+
+#include <sycl/sycl.hpp>
+namespace syclex = sycl::ext::oneapi::experimental;
+
+int main() {
+  sycl::queue q;
+  sycl::device d = q.get_device();
+
+  assert(d.ext_oneapi_can_compile(syclex::source_language::opencl) &&
+         "can_compile(opencl) unexpectedly false");
+
```

Comment:
Add a test that the following constants are defined with their expected values:

* `opencl_c_1_0`
* `opencl_c_1_1`
* `opencl_c_1_2`
* `opencl_c_2_0`
* `opencl_c_3_0`

Reply:
done